app/src/main/java/com/duckduckgo/app/global/UriExtension.kt

```diff
@@ -46,12 +46,24 @@ val Uri.hasIpHost: Boolean
         return baseHost?.matches(ipRegex) ?: false
     }
 
+private const val MOBILE_URL_PREFIX_SHORT = "m."
```

Comment:
Does it make sense to use a list here? I think it may simplify the logic and as a bonus we could add more of these as we find them without needing to rewrite the code. This would give us something along the lines of:
```
private val MOBILE_URL_PREFIXES = listOf("m.", "mobile.")

val Uri.isMobileSite: Boolean
    get() {
        val auth = authority ?: return false
        return MOBILE_URL_PREFIXES.firstOrNull { auth.startsWith(it) } != null
    }

fun Uri.toDesktopUri(): Uri {
    if (!isMobileSite) return this

    val newUrl = MOBILE_URL_PREFIXES.fold(toString()) { url, prefix ->
        url.replaceFirst(prefix, "")
    }
    return Uri.parse(newUrl)
}
```

Reply:
_fancy_! 😄 

This has _slightly_ different than before, but that might be desirable; this code will continually strip mobile prefixes, rather than just doing it once.

So `https://m.mobile.example.com` and `https://mobile.m.example.com` will both resolve to `https://example.com`.

I think that's a desirable behavioural change. 🤔

Reply:
I think that change is ok (and maybe even good, but at least ok 😄) so i've pushed the suggested changes.

Back at you @subsymbolic to RR.